### ISSUE TITLE
Director view and workflows

### ISSUE THREAD

**Issue by RasterboxOff** [cpvalente/ontime#934]
Hello everyone and thank you @cpvalente for guiding me to the right place to give feedback. 

Since the director view was announced for V3, I thought I share some experiences from corporate events and how workflows there might impact the planned view.

#### The usual workflow means that a director calls the upcoming cue in two steps.

- **Step 1** is to prepare the crew for the upcoming scene: "Get ready for Cue 37." often followed by specifics depending on what is important like "Get ready for Cue 37, walk on jingle, Lighting scene 4... etc." this is then followed by 

- **Step 2** – the actual Call to run the the cue e.g. "Cue 37 Go"– something like that.

If it is just about a single cue, I think **a way to "arm" the cue** – that maybe gives the upcoming cue a highlighted border in the cue sheet and operator view, would be a nice option. After the cue is armed it could then be launched in the second step. Even though this is a very common way to run a show, I think the step of arming a cue should be optional. Arming and firing the cue could be controlled by a single button – eg the space bar – that just guides you through the schedule: Ready - Go, Ready - Go, ...

#### Cues within cues
However, there might be more things to call within a running cue. Let's imagine a powerpoint presentation and we know that some time during slide XY we will switch to a video playback and then back. That is a cue within a cue and also something to keep track of that needs to be called by the director "Get ready for video 3 .... and Go!"  

The bigger challenge is to create a way for these – let's call them nested cues – to be represented. It could be done by custom fields and the director has to make sure to have the information right in the correct fields. It could also be something entirely different, like an actual cue within a cue, with its own time and everything. 

#### Additional info within the view
- This bleeds over into the topic of adding additional timers to Ontime and how they might interact, if we go back to our example with a video within a longer presentation: the director wants to know the remaining running time for that video. 

- Speaking of thing the director might want to see: the camera footage. Maybe the option to leave some space in the layout for. a pip finds a valid use case here.

#### So to summarize my two cents on the topic:
- I think it needs the option to go in a two-step manner by arming and firing cues
- the view needs a way to integrate external timers
- relevant elements that need directing might be nested within a cue – this bears a challenge on how to make that failsafe: are custom fields enough to do that? Is a checklist feature needed, where custom fields can be defined as a task within a cue? Does it even need full support for cues within a cue?
- It might not be the end of the pip option after all?

I'm very curious on how that director view will pan out, all your work is amazing.
